cemu/ui/main.py

```diff
@@ -770,8 +805,13 @@ def run(self):
             #
             if emu.use_step_mode:
                 insn = emu.next_instruction(emu.code[start_offset:], start_address)
-                end_address = insn.end
-                info(f"Stepping from {start_address:#x} to {end_address:#x}")
+                if insn is not None:
+                    end_address = insn.end
+                    info(f"Stepping from {start_address:#x} to {end_address:#x}")
+                else:
+                    if emu.pc() == (emu.start_addr + len(emu.code)):
+                        emu.set(EmulatorState.FINISHED)
+                        return
```

Comment:
this is a nice fail-safe. small nit, if disassembly failed, it must be because an error occured, stopping the emulator seemed the right choice regardless 

```suggestion
                if insn is None:
                    emu.set(EmulatorState.FINISHED)
                    return
                else:
                    end_address = insn.end
                    info(f"Stepping from {start_address:#x} to {end_address:#x}")
```